### PR TITLE
Pass TOML file to action, use compare-locales to parse the configuration

### PR DESCRIPTION
**Pull request by flodolo**
## Docker

- Removed unnecessary requirements, updated to Python 3.10

## Action Configuration

- Pass `toml_path` (path to the TOML config file) instead of a project-name.
- Renamed `dst` to `dest` for clarity

## Entrypoint Script

- Use `compare-locales` to parse the configuration file and copy files.
- Always copy files without comparing them (git will take care of adding files if changed).
- Drop check for l10n-release.toml
- Removed `git` commands. Commit is already part of a separate workflow, no point in committing files in this action.

### CODE REVIEW

**Comment by flodolo** [comment]
I've been testing this on my fork
https://github.com/flodolo/firefox-android/actions/runs/4035607891/jobs/6937620391
https://github.com/flodolo/firefox-android/actions/runs/4035602279/jobs/6937611418

File: README.md [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-      - name: "Checkout Master Branch"
-        uses: actions/checkout@v2
+      - name: Checkout default branch
+        uses: actions/checkout@v3
         with:
-          path: main
-          ref: master
-      - name: "Checkout Beta Branch"
-        uses: actions/checkout@v2
+          path: default
```
Comment (gabrielluong) [inline README.md]: Let's keep the explicit `path` and `ref` here, but change it to the main branch

File: README.md [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
-        uses: st3fan/sync-strings-action@main
+      - name: Sync strings
+        uses: mozilla-mobile/sync-strings-action@main
         with:
-          project: android-components
-          src: main
-          dst: beta
-
+          toml_path: android-components/l10n.toml
+          src: default
```
Comment (gabrielluong) [inline README.md]:
```suggestion
          src: main
```

File: src/sync-strings.py [hunk 3]
```diff
@@ -1,108 +1,86 @@
...
-    with open(os.path.join(repo_path, project_path, "l10n.toml")) as main_toml_fp:
-        main_l10n_toml = tomlkit.loads(main_toml_fp.read())
-        for locale in locales:
-            pathname = main_l10n_toml["paths"][0]["l10n"].replace("{android_locale}", android_locale(locale))
-            for strings_xml_path in glob.glob(os.path.join(repo_path, project_path, pathname), recursive=True):
-                yield os.path.relpath(strings_xml_path, repo_path)
+def main():
+    src_repo_path = os.environ.get("INPUT_SRC")
+    if src_repo_path is None:
+        sys.exit("Empty 'src' parameter. Exiting.")
```
Comment (gabrielluong) [inline src/sync-strings.py:50]: Is this an appropriate exit code? Reading the docs https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions indicates that it needs to be any integer but 0. So, I think it's best if we kept with the existing `exit(1)` and `print(string)`

Comment (flodolo) [reply src/sync-strings.py:50]: I believe it is correct ([sys.exit docs](https://docs.python.org/3/library/sys.html#sys.exit))

> If another type of object is passed, None is equivalent to passing zero, and any other object is printed to stderr and results in an exit code of 1. In particular, sys.exit("some error message") is a quick way to exit a program when an error occurs.

Let me know if you still want to change it.

Comment (flodolo) [reply src/sync-strings.py:50]: Side note: I've been using this approach without issues in GitHub workflows.

File: README.md [hunk 4]
```diff
@@ -1,30 +1,26 @@
...
 
-This GitHub Action syncs strings between two checked out copies of an Android project. It requires that these projects have a a `l10n.toml` and optionally an `l10n-release.toml` file in their root.
+This GitHub Action syncs strings between two checked out copies of an Android project. It requires that these projects have a a `l10n.toml` project configuration.
 
 Example usage in a _GitHub Workflow_ file:
 
 ```
-      - name: "Checkout Master Branch"
-        uses: actions/checkout@v2
+      - name: Checkout default branch
```
Comment (gabrielluong) [inline README.md]:
```suggestion
      - name: Checkout main branch
```

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**